OneSignalSDK/unittest/src/test/java/com/test/onesignal/DatabaseRunner.java

```diff
@@ -483,4 +479,207 @@ public void shouldUpgradeDbFromV7ToV8OutcomesTable() throws JSONException {
         assertEquals(new JSONArray(), outcomeSaved.getIamIds());
         assertEquals(OSInfluenceType.UNATTRIBUTED, outcomeSaved.getIamInfluenceType());
     }
+
+    @Test
+    public void shouldUpgradeDbFromV3ToV8UniqueOutcomeTable() {
```

Comment:
Even though this test does not fail it still prints the following as an error in the log which needs to be fixed.
```java
Caused by: java.util.concurrent.ExecutionException: com.almworks.sqlite4java.SQLiteException: [1] DB[2] prepare() INSERT INTO outcome_backup SELECT _id,session,notification_ids,name,timestamp FROM outcome; [no such column: session]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.google.common.util.concurrent.Uninterruptibles.getUninterruptibly(Uninterruptibles.java:199)
	at org.robolectric.shadows.ShadowSQLiteConnection$Connections.getFuture(ShadowSQLiteConnection.java:734)
	at org.robolectric.shadows.ShadowSQLiteConnection$Connections.execute(ShadowSQLiteConnection.java:728)
	at org.robolectric.shadows.ShadowSQLiteConnection$Connections.prepareStatement(ShadowSQLiteConnection.java:420)
	at org.robolectric.shadows.ShadowSQLiteConnection.nativePrepareStatement(ShadowSQLiteConnection.java:90)
	at android.database.sqlite.SQLiteConnection.nativePrepareStatement(SQLiteConnection.java)
	at android.database.sqlite.SQLiteConnection.$$robo$$android_database_sqlite_SQLiteConnection$acquirePreparedStatement(SQLiteConnection.java:889)
	at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java)
	at android.database.sqlite.SQLiteConnection.$$robo$$android_database_sqlite_SQLiteConnection$prepare(SQLiteConnection.java:500)
	at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java)
	at android.database.sqlite.SQLiteSession.$$robo$$android_database_sqlite_SQLiteSession$prepare(SQLiteSession.java:588)
	at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java)
	at android.database.sqlite.SQLiteProgram.$$robo$$android_database_sqlite_SQLiteProgram$__constructor__(SQLiteProgram.java:58)
	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java)
	at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java)
	at android.database.sqlite.SQLiteDatabase.$$robo$$android_database_sqlite_SQLiteDatabase$executeSql(SQLiteDatabase.java:1677)
	at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java)
	at android.database.sqlite.SQLiteDatabase.$$robo$$android_database_sqlite_SQLiteDatabase$execSQL(SQLiteDatabase.java:1608)
	at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java)
	at com.onesignal.outcomes.OSOutcomeTableProvider.$$robo$$com_onesignal_outcomes_OSOutcomeTableProvider$upgradeOutcomeTableRevision1To2(OSOutcomeTableProvider.java:71)
	at com.onesignal.outcomes.OSOutcomeTableProvider.upgradeOutcomeTableRevision1To2(OSOutcomeTableProvider.java)
	at com.onesignal.OneSignalDbHelper.$$robo$$com_onesignal_OneSignalDbHelper$upgradeFromV5ToV6(OneSignalDbHelper.java:263)
	at com.onesignal.OneSignalDbHelper.upgradeFromV5ToV6(OneSignalDbHelper.java)
	at com.onesignal.OneSignalDbHelper.$$robo$$com_onesignal_OneSignalDbHelper$upgradeToV5(OneSignalDbHelper.java:256)
	at com.onesignal.OneSignalDbHelper.upgradeToV5(OneSignalDbHelper.java)
	at com.onesignal.OneSignalDbHelper.$$robo$$com_onesignal_OneSignalDbHelper$internalOnUpgrade(OneSignalDbHelper.java:198)
	at com.onesignal.OneSignalDbHelper.internalOnUpgrade(OneSignalDbHelper.java)
	at com.onesignal.OneSignalDbHelper.$$robo$$com_onesignal_OneSignalDbHelper$onUpgrade(OneSignalDbHelper.java:179)
	at com.onesignal.OneSignalDbHelper.onUpgrade(OneSignalDbHelper.java)
	at android.database.sqlite.SQLiteOpenHelper.$$robo$$android_database_sqlite_SQLiteOpenHelper$getDatabaseLocked(SQLiteOpenHelper.java:299)
	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java)
	at android.database.sqlite.SQLiteOpenHelper.$$robo$$android_database_sqlite_SQLiteOpenHelper$getWritableDatabase(SQLiteOpenHelper.java:194)
	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java)
	at com.onesignal.OneSignalDbHelper.$$robo$$com_onesignal_OneSignalDbHelper$getSQLiteDatabaseWithRetries(OneSignalDbHelper.java:155)
	at com.onesignal.OneSignalDbHelper.getSQLiteDatabaseWithRetries(OneSignalDbHelper.java)
	at com.onesignal.MockOneSignalDBHelper.$$robo$$com_onesignal_MockOneSignalDBHelper$getSQLiteDatabaseWithRetries(MockOneSignalDBHelper.java:22)
	at com.onesignal.MockOneSignalDBHelper.getSQLiteDatabaseWithRetries(MockOneSignalDBHelper.java)
	at com.test.onesignal.DatabaseRunner.shouldUpgradeDbFromV3ToV8UniqueOutcomeTable(DatabaseRunner.java:508)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	... 13 more
```